src/api/procedures/inviteExternalAgent.ts

```diff
@@ -117,16 +119,27 @@ export async function prepareInviteExternalAgent(
     context
   );
 
+  let postTransactionAuthorization: MaybePostTransactionValue<Authorization>;
   let rawAuthorizationData;
 
   if (permissions instanceof KnownPermissionGroup || permissions instanceof CustomPermissionGroup) {
+    postTransactionAuthorization = {
+      type: AuthorizationType.BecomeAgent,
+      value: permissions,
+    };
     rawAuthorizationData = authorizationDataResolver(permissions, context);
   } else {
     // We know this procedure returns a PostTransactionValue, so this assertion is necessary
     const createGroupResult = (await this.addProcedure(createGroup(), {
       ticker,
       permissions,
     })) as PostTransactionValue<CustomPermissionGroup>;
+    postTransactionAuthorization = createGroupResult.transform(customPermissionGroup => {
+      return {
+        type: AuthorizationType.BecomeAgent,
+        value: customPermissionGroup,
+      };
+    });
     rawAuthorizationData = createGroupResult.transform(customPermissionGroup =>
```

Comment:
I was thinking more of something like "chaining" the transformations. So instead of having `authorizationDataResolver` that goes directly from permission to authorizationData, you split it into two steps. One that goes from permission to Authorization, and then just `authorizationToAuthorizationData`

You can do the same with transform, using `transform` on the result of the first `transform` call to get the second result

Reply:
I see. That does clean up the file a bit.

The mock in the test case is a bit more complicated, but I think its worth it.